### PR TITLE
Fixes reference to undefined

### PR DESCRIPTION
**Pull request by jmnsf**
Fixes https://github.com/webdriverio/wdio-sync/issues/51

### CODE REVIEW

**Comment by christian-bromann** [comment]
@jmnsf could you please add unit tests for `is$$Result`?

**Comment by jmnsf** [comment]
@christian-bromann I would, normally, but the function isn't exported. Also, I would wager that an array of `undefined`s might be symptom of a bug elsewhere.

Honestly this is more of a hotfix since the exception is breaking my test suite through no fault of my own. Worse, it's a package that's depended on by `wdio-mocha-framework` which makes it a bit of a pain to pull into fix through git installs...

Feel free to edit this PR and add tests that handle this, but I would really appreciate a fast merge on what is essentially a defensive programming patch.

**Comment by christian-bromann** [comment]
> but the function isn't exported

Let's export it into a helper file then.

> I would really appreciate a fast merge on what is essentially a defensive programming patch.

I agree with you .. however I see a value in having this covered in unit tests. I can take care of it but you will have to wait until I have time to work on that

**Comment by jmnsf** [comment]
@christian-bromann then if you don't get to it in the meantime, I'll have a look later today

**Comment by jmnsf** [comment]
@christian-bromann Done. Had to update the rewrite plugin because of a bug.

File: lib/results.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+let isElements = function (result) {
```
Comment (christian-bromann) [inline lib/results.js]: let's do
```js
exports const isElements = ...
```

File: lib/results.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    return (
+        typeof result.selector === 'string' &&
+        Array.isArray(result.value) && result.value.length &&
+        typeof result.value[0].ELEMENT !== 'undefined'
+    )
+}
+
+let is$$ = function (result) {
```
Comment (christian-bromann) [inline lib/results.js]: same here

File: index.js [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import Future from 'fibers/future'
 import assign from 'object.assign'
+
+import * as results from './lib/results'
```
Comment (christian-bromann) [inline index.js]: I find `results` confusing, let's import these helpers as they are:
```js
import { is$$, isElements } from './lib/results'
```

**Comment by jmnsf** [comment]
@christian-bromann ready

**Comment by christian-bromann** [comment]
Will release this version with updated framework adapter later today

**Comment by jmnsf** [comment]
Awesome, thank you ;)

**Comment by jmnsf** [comment]
@christian-bromann hey, could we have this published soon, we're stuck on older package versions until then :)

**Pull request merged**